xrpl/core/binarycodec/types/account_id.py

```diff
@@ -59,7 +65,16 @@ def from_value(cls: Type[AccountID], value: str) -> AccountID:
         if _HEX_REGEX.fullmatch(value):
             return cls(bytes.fromhex(value))
         # base58 case
-        return cls(decode_classic_address(value))
+        if is_valid_classic_address(value):
+            return cls(decode_classic_address(value))
+        if is_valid_xaddress(value):
+            classic_address, _, _ = xaddress_to_classic_address(value)
```

Comment:
Do we handle the Destination/Source Tags somewhere else in the binarycodec?

Reply:
Yep, happens in https://github.com/XRPLF/xrpl-py/blob/master/xrpl/core/binarycodec/types/serialized_dict.py